internal/controller/pipelinerollout_controller_test.go

```diff
@@ -183,8 +189,19 @@ var _ = Describe("PipelineRollout Controller", Ordered, func() {
 				return updatedChildResource.Spec, nil
 			}, timeout, interval).Should(Equal(pipelineSpec))
 
+			/*By("Verifying the Numaflow Pipeline lifecycle desiredPhase is 'Paused'")
+			Eventually(func() (string, error) {
+				updatedChildResource := &numaflowv1.Pipeline{}
+				err := k8sClient.Get(ctx, resourceLookupKey, updatedChildResource)
+				if err != nil {
+					return "", err
+				}
+				return string(updatedChildResource.Spec.Lifecycle.DesiredPhase), nil
+			}, timeout, interval).Should(Equal("Paused"))
+
 			By("Verifying that the PipelineRollout Status Phase is Deployed and ObservedGeneration matches Generation")
-			verifyStatusPhase(ctx, apiv1.PipelineRolloutGroupVersionKind, namespace, pipelineRolloutName, apiv1.PhaseDeployed)
+			verifyStatusPhase(ctx, apiv1.PipelineRolloutGroupVersionKind, namespace, pipelineRolloutName, apiv1.PhaseDeployed)*/
+
```

Comment:
The unit test is currently testing with the `DataLossPrevention` feature flag turned off. 

Once we try to test with this feature on, I am starting to think much of our unit testing should actually become e2e tests instead. The idea of each reconciler being a unit unto itself without cross dependencies with the other reconcilers is becoming less the case. If I were to try to verify that the Pipeline gets fully deployed from being paused to becoming running and applying the new spec, I really need to have a Numaflow Controller running as well.